master/buildbot/locks.py

```diff
@@ -96,15 +99,16 @@ def isAvailable(self, requester, access):
 
         if access.mode == 'counting':
             # Wants counting access
-            return num_excl == 0 and num_counting + len(ahead) < self.maxCount \
+            return num_excl == 0 \
+                and num_counting + len(ahead) + access.count <= self.maxCount \
                 and all([w[1].mode == 'counting' for w in ahead])
         # else Wants exclusive access
         return num_excl == 0 and num_counting == 0 and not ahead
```

Comment:
that's already existing code, but the comment still applies

Reply:
ok

Reply:
done